dashboard/modules/job/cli.py

```diff
@@ -270,18 +278,29 @@ def submit(
         runtime_env_json=runtime_env_json,
         working_dir=working_dir,
     )
-    job_id = client.submit_job(
-        entrypoint=list2cmdline(entrypoint),
-        submission_id=submission_id,
-        runtime_env=final_runtime_env,
-        metadata=metadata_json,
-        entrypoint_num_cpus=entrypoint_num_cpus,
-        entrypoint_num_gpus=entrypoint_num_gpus,
-        entrypoint_memory=entrypoint_memory,
-        entrypoint_resources=entrypoint_resources,
-    )
 
-    _log_big_success_msg(f"Job '{job_id}' submitted successfully")
+    try:
+        job_id = client.submit_job(
+            entrypoint=list2cmdline(entrypoint),
+            submission_id=submission_id,
+            runtime_env=final_runtime_env,
+            metadata=metadata_json,
+            entrypoint_num_cpus=entrypoint_num_cpus,
+            entrypoint_num_gpus=entrypoint_num_gpus,
+            entrypoint_memory=entrypoint_memory,
+            entrypoint_resources=entrypoint_resources,
+        )
+
+        _log_big_success_msg(f"Job '{job_id}' submitted successfully")
+    except Exception as e:
```

Comment:
This feels a little hacky to me. 

It would be better if `client.submit_job` returned a valid 404 response that we can check, but instead it raises an exception:

https://github.com/ray-project/ray/blob/c74837002ff1ce4ed7535ba063f29c70f70f346f/dashboard/modules/job/sdk.py#L251-L254

I tried to change `client.submit_job` but then I realized this would be a breaking change to a stable API in the SDK and didn't bother.